onnxruntime/core/providers/qnn/qnn_execution_provider.cc

```diff
@@ -231,10 +231,21 @@ QNNExecutionProvider::GetSupportedNodes(const GraphViewer& graph_viewer,
     initializer_input_lookup.emplace(graph_ini.first);
   }
 
+  // Util function that initializes a table that maps a graph input or output name to its index.
+  auto init_input_output_index_map = [](std::unordered_map<std::string, size_t>& index_map,
+                                        const std::vector<const NodeArg*>& node_args) {
+    const size_t num_args = node_args.size();
+    for (size_t i = 0; i < num_args; i++) {
+      index_map.emplace(node_args[i]->Name(), i);
+    }
+  };
+
   std::unordered_map<std::string, size_t> model_input_index_map;
+  init_input_output_index_map(model_input_index_map, graph_viewer.GetInputs());  // GetInputs excludes initializers.
+
   std::unordered_map<std::string, size_t> model_output_index_map;
-  std::unordered_map<std::string, qnn::OnnxTensorInfo> inputs_info;
-  std::unordered_map<std::string, qnn::OnnxTensorInfo> outputs_info;
+  init_input_output_index_map(model_output_index_map, graph_viewer.GetOutputs());
```

Comment:
These changes allow us to properly detect which tensors are graph inputs or outputs during validation (GetCapability). We need this to be able to reject ArgMax/ArgMin ops which produce an output.

Previously, we didn't need to care whether a tensor was a graph input/output for validation purposes.

Reply:
Is there any missing commits for this PR? Why I can't find the place init_input_output_index_map get used?

---
In reply to: [1265637386](https://github.com/microsoft/onnxruntime/pull/16731#discussion_r1265637386) [](ancestors = 1265637386)

Reply:
No commits missing. Here's the diff on github:
![pr_screenshot](https://github.com/microsoft/onnxruntime/assets/19691973/84e088b5-ae91-4fea-9d74-a7fb2bfd6dbb)

Reply:
I see. Thanks.

---
In reply to: [1265673357](https://github.com/microsoft/onnxruntime/pull/16731#discussion_r1265673357) [](ancestors = 1265673357)